0000-settlement-engine/0000-settlement-engine.md

```diff
@@ -0,0 +1,353 @@
+
+##### Request
+
+```http
+POST /accounts/:id HTTP/1.1
+```
+
+##### Response
+
+```http
+HTTP/1.1 201 CREATED
+```
+
+#### Perform outgoing settlement
+
+Reconcile a liability owed to the peer for the given amount.
+
+Note that the settlement engine MAY accrue owed settlements without settling immediately due to varying arrangements, implementations, or conditions. The settlement engine SHOULD persist amounts owed to the peer that have yet to be settled so they can be settled later.
+
+The accounting system MUST be the only entity invoking this instruction and MUST preemptively debit this amount from the accounts payable, or some liability account tracking a balance owed to the peer, to ensure accurate accounting.
+
+##### Request
+
+```http
+POST /accounts/:id/settle HTTP/1.1
+Accept: application/json
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** to settle
+>
+> - If the amount is `0`, the settlement engine MAY retry failed or queued settlements.
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+
+> **[`Quantity`](#quantity-type)** enqueued to settle
+>
+> - This response does not guarantee that a settlement was executed.
+> - If the settlement engine uses a unit less precise than the accounting system's unit, in order to prevent the two systems getting out-of-sync, it must indicate the amount it queued so the accounting system can correctly track the amount leftover.
+
+#### Handle incoming request
+
+Respond to an incoming message from the given peer's settlement engine. The transport MUST be the only entity invoking this instruction.
+
+##### Request
+
+```http
+POST /accounts/:id/handleMessage HTTP/1.1
+Accept: application/octet-stream
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of message&gt;_
+
+##### Response
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/octet-stream
+```
+
+> _&lt;raw bytes of response&gt;_
+
+#### Complete specification
+
+<details>
+  <summary>The full settlement engine HTTP API, including all response codes, error codes, and additional endpoints for manual management are available in this Swagger definition.</summary>
+
+```
+TODO Embed the definition here
+```
+
+</details>
+
+Settlement engine implementations are RECOMMENDED to implement the standardized endpoints for manual management noted above.
+
+#### Additional RPCs
+
+Settlement engine implementations MAY expose additional, non-standard endpoints for manual operations or configuration.
+
+For example, settlement engines that use payment channels may expose functionality to manually open, fund, and close channels.
+
+### Accounting System HTTP API
+
+#### Credit incoming settlement
```

Comment:
Could you have a counterpart to this where the settlement engine notifies the accounting system that it sent some money? This could be useful when sending isn't a direct consequence of a message from the accounting system, for instance if the leftover rounded off amounts have just been settled after some timeout, or if a manual call to the settlement engine triggered a settlement

Reply:
What would the accounting system do with this information?

Reply:
I suppose if you wanted to do a force-settlement you could tell the accounting system to ask the settlement engine to do a settlement

Reply:
Resolving since for manual settlements, the accounting system may implement their own endpoints that operators could call so as to not break the double entry bookkeeping.

(Feel free to re-open if there are other use cases for this/further discussion is necessary).